### PR TITLE
Es6 refactor use file plugin

### PR DESCRIPTION
**Pull request by rachel-ftw**

### CODE REVIEW

File: lib/UseFilePlugin.js [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function UseFilePlugin(source, filename, target) {
-	this.source = source;
-	this.filename = filename;
-	this.target = target;
-}
-module.exports = UseFilePlugin;
+const assign = require("object-assign");
```
Comment (TheLarkInn) [inline lib/UseFilePlugin.js]: Ooo use Object.assign() now instead of `assign()`. Same syntax and etc. But we can get rid of this tiny package once all refactored out.

File: lib/UseFilePlugin.js [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+	apply(resolver) {
+		const filename = this.filename;
+		const target = this.target;
+		resolver.plugin(this.source, (request, callback) => {
+			const filePath = resolver.join(request.path, filename);
+			const obj = assign({}, request, {
+				path: filePath,
+				relativePath: request.relativePath && resolver.join(request.relativePath, filename)
+			});
+			resolver.doResolve(target, obj, "using path: " + filePath, callback);
```
Comment (TheLarkInn) [inline lib/UseFilePlugin.js]: Let's do a template string here. `using path ${}`

File: lib/UseFilePlugin.js (hunk 1 above)
Comment (rachel-ftw) [inline lib/UseFilePlugin.js]: √, I'll work thought this

File: lib/UseFilePlugin.js (hunk 2 above)
Comment (rachel-ftw) [inline lib/UseFilePlugin.js]: √

File: lib/UseFilePlugin.js (hunk 1 above)
Comment (TheLarkInn) [inline lib/UseFilePlugin.js]: Sure, for now just remove from this file for this PR, and then do the rest in a separate PR.

File: lib/UseFilePlugin.js [hunk 3]
```diff
@@ -1,23 +1,27 @@
+"use strict";
```
Comment (sokra) [inline lib/UseFilePlugin.js]: use strict after comment

**Comment by ooflorent** [comment]
Thank you for your PR but it appears that the file was previously converted to ES6. I'm closing the PR for this reason.
